node/src/utils/specimen.rs

```diff
@@ -0,0 +1,951 @@
+                }
+                ExecutableDeployItemDiscriminants::StoredContractByHash => {
+                    ExecutableDeployItem::StoredContractByHash {
+                        hash: LargestSpecimen::largest_specimen(estimator, cache),
+                        entry_point: largest_contract_entry_point(estimator),
+                        args: LargestSpecimen::largest_specimen(estimator, cache),
+                    }
+                }
+                ExecutableDeployItemDiscriminants::StoredContractByName => {
+                    ExecutableDeployItem::StoredContractByName {
+                        name: largest_contract_name(estimator),
+                        entry_point: largest_contract_entry_point(estimator),
+                        args: LargestSpecimen::largest_specimen(estimator, cache),
+                    }
+                }
+                ExecutableDeployItemDiscriminants::StoredVersionedContractByHash => {
+                    ExecutableDeployItem::StoredVersionedContractByHash {
+                        hash: LargestSpecimen::largest_specimen(estimator, cache),
+                        version: LargestSpecimen::largest_specimen(estimator, cache),
+                        entry_point: largest_contract_entry_point(estimator),
+                        args: LargestSpecimen::largest_specimen(estimator, cache),
+                    }
+                }
+                ExecutableDeployItemDiscriminants::StoredVersionedContractByName => {
+                    ExecutableDeployItem::StoredVersionedContractByName {
+                        name: largest_contract_name(estimator),
+                        version: LargestSpecimen::largest_specimen(estimator, cache),
+                        entry_point: largest_contract_entry_point(estimator),
+                        args: LargestSpecimen::largest_specimen(estimator, cache),
+                    }
+                }
+                ExecutableDeployItemDiscriminants::Transfer => ExecutableDeployItem::Transfer {
+                    args: LargestSpecimen::largest_specimen(estimator, cache),
+                },
+            }
+        })
+    }
+}
+
+impl LargestSpecimen for U512 {
+    fn largest_specimen<E: SizeEstimator>(_estimator: &E, _cache: &mut Cache) -> Self {
+        U512::max_value()
+    }
+}
+
+impl LargestSpecimen for ContractHash {
+    fn largest_specimen<E: SizeEstimator>(estimator: &E, cache: &mut Cache) -> Self {
+        ContractHash::new([LargestSpecimen::largest_specimen(estimator, cache); KEY_HASH_LENGTH])
+    }
+}
+
+impl LargestSpecimen for ContractPackageHash {
+    fn largest_specimen<E: SizeEstimator>(estimator: &E, cache: &mut Cache) -> Self {
+        ContractPackageHash::new(
+            [LargestSpecimen::largest_specimen(estimator, cache); KEY_HASH_LENGTH],
+        )
+    }
+}
+
+impl LargestSpecimen for RuntimeArgs {
+    fn largest_specimen<E: SizeEstimator>(_estimator: &E, _cache: &mut Cache) -> Self {
+        Default::default()
```

Comment:
An empty vec doesn't seem right here?

We have `[deploys.session_args_max_length]` and `[deploys.payment_args_max_length]` which determine the maximum sizes (bytesrepr-encoded) of the `RuntimeArgs` in a deploy.

Reply:
Has been removed, as we don't need it if we are just generating the maximum deploy size as bytecode.